### PR TITLE
Working Android screen Recording

### PR DESCRIPTION
**Pull request by rdurish**
**Things Performed:**
- Updated Source and Example to dotnet 8
- Added Android Screen Recording to Src
- Added MediaElement to see screen video after recording.
- Updated Read Me with implementation instructions for Android.

**Might Want to Change:**
- Currently the only way I could get this to work was AutoFac for dependency Injection.  I would much rather it use the standard Injection but I couldn't get it to work, as the object has to remain the same across different android states...

### CODE REVIEW

**Comment by jfversluis** [comment]
This is amazing! Thank you so much for this!

~Didn't look in detail yet, but do we need _any_ dependency injection framework? I would love to not introduce any additional dependencies for this...~ Looks like its just for the sample? That's alright... But why don't we just use the built-in dependency injection?

Also seems like the GitHub Actions need an update :D

File: samples/ScreenRecordingSample/Platforms/Android/AndroidManifest.xml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android">
-	<application android:allowBackup="true" android:icon="@mipmap/appicon" android:roundIcon="@mipmap/appicon_round" android:supportsRtl="true"></application>
+	<application android:allowBackup="true" android:icon="@mipmap/appicon" android:roundIcon="@mipmap/appicon_round" android:supportsRtl="true" android:requestLegacyExternalStorage="true" android:preserveLegacyExternalStorage="true">
+		<service android:name="Plugin.Maui.ScreenRecording.ScreenRecordingImplementation.ScreenRecordingService" android:exported="false" android:foregroundServiceType="mediaProjection" />
+	</application>
+	<uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+	<uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
+	<uses-permission android:name="android.permission.RECORD_AUDIO" />
+	<uses-permission android:name="android.permission.FOREGROUND_SERVICE" />
```
Comment (jfversluis) [inline samples/ScreenRecordingSample/Platforms/Android/AndroidManifest.xml:9]: For API 33 it works! But for API 34 I see this exception, seems we need more permissions? But not sure which...

> Exception has occurred: Java.Lang.SecurityException
Starting FGS with type mediaProjection callerApp=ProcessRecord{786d26 24066:com.companyname.screenrecordingsample/u0a182} targetSDK=34 requires permissions: all of the permissions allOf=true [android.permission.FOREGROUND_SERVICE_MEDIA_PROJECTION] any of the permissions allOf=false [android.permission.CAPTURE_VIDEO_OUTPUT, android:project_media] 

I added `android.permission.FOREGROUND_SERVICE_MEDIA_PROJECTION` but then it crashes when I press start recording without any good information about why it crashes.

Comment (rdurish) [reply samples/ScreenRecordingSample/Platforms/Android/AndroidManifest.xml:9]: I'm an apple guy lol, my dev android only goes to 31, but that is good to know!  :)  It works on there.

**Comment by rdurish** [comment]
> This is amazing! Thank you so much for this!
> 
> ~Didn't look in detail yet, but do we need _any_ dependency injection framework? I would love to not introduce any additional dependencies for this...~ Looks like its just for the sample? That's alright... But why don't we just use the built-in dependency injection?
> 
> Also seems like the GitHub Actions need an update :D

I couldn't for the life of me get it to work without using that plugin.  I spent a ton of time, honestly not sure what is different with AutoFac versus the norm.  I tried to do Singleton, but when it would go from the Maui to the android code it would be a fresh instance and all the objects would be null and it wouldn't work.

File: samples/ScreenRecordingSample/Platforms/Android/AndroidManifest.xml (hunk 1 above)
Comment (jfversluis) [reply samples/ScreenRecordingSample/Platforms/Android/AndroidManifest.xml:9]: `MediaProjection` is null in the Android implementation on line 164

Comment (rdurish) [reply samples/ScreenRecordingSample/Platforms/Android/AndroidManifest.xml:9]: I ran it before I pushed.  I will verify when I get home.

Comment (jfversluis) [reply samples/ScreenRecordingSample/Platforms/Android/AndroidManifest.xml:9]: Yeah not sure what is going on for API 34, but seems we need an extra check there maybe

**Comment by jfversluis** [comment]
> I couldn't for the life of me get it to work without using that plugin. I spent a ton of time, honestly not sure what is different with AutoFac versus the norm. I tried to do Singleton, but when it would go from the Maui to the android code it would be a fresh instance and all the objects would be null and it wouldn't work.

Check out the last commit. I removed AutoFac and reworked some things. This way:

* We will only ask for the permission when the actual recording is started. I guess people might get nervous when you immediately prompt for recording their screen if it's not really clear that the app should be doing that and it's only for support purposes for instance.
* It is more self-contained; no need to let the user add stuff in their MainActivity, we all do that for them
* No redundant Setup() method for platforms that don't need it
* We have no extra dependencies!

It seems to all work fine on my end but please check it on yours!

**Comment by rdurish** [comment]
Great!  I will take a look tonight!

**Comment by rdurish** [comment]
That works on my device.  What did you have to do to fix it on your API 34?

**Comment by jfversluis** [comment]
@rdurish this is independent of API 34, that is still not working 😅 

Got a bit clearer exception message now though

> java.lang.IllegalStateException: Must register a callback before starting capture, to manage resources in response to MediaProjection states.

**Comment by jfversluis** [comment]
OK see that last commit. We needed to add some callback for API 34. Now we just need to implement some actual cleanup in there probably 😄

**Comment by jfversluis** [comment]
OK I think we're almost there. We probably want to clean up some more warnings and stuff, but I think we're in pretty good shape here!

~I do notice some entirely green frames for each recording. Any idea why that might be?~ looks like that is a non-issue. Testing on Windows and another emulator I don't see it.

**Comment by jfversluis** [comment]
OK I think we're ready for a first preview release. I could not have done this without you @rdurish ! Thank you so much for this!

**Pull request merged**